### PR TITLE
Karsten/dependencies

### PR DESCRIPTION
**Pull request by blacklab**
We drop git submodules and use a Makefile to manage dependencies.

### CODE REVIEW

**Comment by boxdot** [comment]
We've also 
1. removed old renderer code,
2. download and build dependencies just on `make`, i.e. `make bootstrap` does not have to be called before anymore.

File: deps.mk [hunk 1]
```diff
@@ -1,4 +1,4 @@
-.PHONY: build-deps dep-a dep-b
+.PHONY: build-deps dep-a dep-b assimp catch docopt ThreadPool
```
Comment (blacklab) [inline deps.mk]: Remove dep-a and dep-b ; )

File: Makefile [hunk 2]
```diff
@@ -19,16 +19,15 @@ raycaster: raycaster.o lib/types.o lib/triangle.o
 raytracer: main.o raytracer.o lib/types.o lib/triangle.o
 pathtracer: main.o pathtracer.o lib/types.o lib/triangle.o
 
-main.o: CXXFLAGS += -Ivendor/ThreadPool -Ivendor/docopt.cpp -pthread
+main.o: CXXFLAGS += -Ivendor/ThreadPool -Ivendor/docopt -pthread
 
-bootstrap: build-deps
+bootstrap: vendor/.last-build
```
Comment (blacklab) [inline Makefile]: How does this work? There is no target for `vendor/.last-build`.

Comment (boxdot) [inline Makefile]: There is in deps.mk. It is included.

**Pull request merged**